### ISSUE TITLE
The implementation of fixed LLM

### ISSUE THREAD

**Issue by SnowCharmQ** [RamyaLab/pluralistic-alignment#8]
Hi, thank you for your excellent work on PAL — I have been following it closely.

While reproducing the code, I noticed a small issue: when running `main_pal_b_fixed_llm.py`, the LLM is still being updated during training, which results in a very large number of trainable parameters.

After some debugging, I found the cause in `pal_rm_b/learner.py` (Line 73). The code currently only uses `self.llm.eval()`, but this does not actually freeze the LLM’s parameters. I tried adding the following implementation:

```python3
for p in self.llm.parameters():
    p.requires_grad = False
```

With this change, the LLM parameters are successfully frozen.

I would like to know if this is the correct way to handle the issue, or whether you had a different design intention here.

Thanks!

**Comment by ChenDaiwei-99** [comment]
Hi, thank you so much for taking the time to reproduce our code and for this valuable feedback!

You're absolutely right to point this out. I can see how the current implementation might be confusing. Let me clarify our approach:

**_In `main_pal_b_fixed_llm.py`, we do freeze the LLM parameters, but we handle it at the optimizer level rather than at the parameter level. Specifically, at `main_pal_b_fixed_llm.py` [L52](https://github.com/RamyaLab/pluralistic-alignment/blob/54feb2f423cbb6cf8eb50d0c3be2038ecf6a8b1f/main_pal_b_fix_llm.py#L52) and the corresponding `pal_rm_b/learner.py` [L70-L71](https://github.com/RamyaLab/pluralistic-alignment/blob/54feb2f423cbb6cf8eb50d0c3be2038ecf6a8b1f/src/pal_rm_b/learner.py#L70-L71), we exclude the LLM parameters from the optimizer's parameter list. This achieves the same effect of preventing gradient updates to the LLM._**

However, I agree that your approach would improve the code readability:
```python
for p in self.llm.parameters():
    p.requires_grad = False
```

This is more explicit and makes the intent much clearer to anyone reading the code. We'll update our implementation to incorporate this improvement. Thank you very much. 🤗
